### PR TITLE
Implement java mock service generator

### PR DESCRIPTION
**Pull request by shinfan**
- Implement mock service generator under MVVM
  - View models
     - MockServiceView
     - MockServiceImplView
     - MockGrpcMethodView
  - Views
     - mock_service.snip
     - mock_service_impl.snip

- Fix output file name

- Testing
  - Baseline test
  - Manual sanity test with one API

### CODE REVIEW

File: src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceNamer.java [hunk 1]
```diff
@@ -41,6 +42,11 @@ public JavaSurfaceNamer() {
   }
 
   @Override
+  public String getFullFilePath(String path, String name) {
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceNamer.java]: I think the name is too generic for the intent here. Maybe rename:

* method -> getSourceFilePath
* name (parameter) -> className

**Comment by garrettjonesgoogle** [comment]
I only had one issue to mention, otherwise looks good.

File: src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceNamer.java (hunk 1 above)
Comment (shinfan) [inline src/main/java/com/google/api/codegen/transformer/java/JavaSurfaceNamer.java]: Done

**Comment by garrettjonesgoogle** [comment]
LGTM assuming tests pass. Your choice if you want to wait for others to look at it.

**Comment by bjwatson** [comment]
No issues jump out at me

**Comment by michaelbausor** [comment]
Same from me, looks good

**Pull request merged**